crates/wash-lib/src/cli/par.rs

```diff
@@ -0,0 +1,179 @@
+    pub capid: String,
+    pub vendor: String,
+    pub revision: Option<i32>,
+    pub version: Option<String>,
+    pub schema: Option<PathBuf>,
+    pub issuer: Option<String>,
+    pub subject: Option<String>,
+    pub name: String,
+    pub directory: Option<PathBuf>,
+    pub arch: String,
+    pub binary: String,
+    pub destination: Option<String>,
+    pub compress: bool,
+    pub disable_keygen: bool,
+}
+pub struct ParInsertArgs {
+    pub archive: String,
+    pub arch: String,
+    pub binary: String,
+    pub directory: Option<PathBuf>,
+    pub issuer: Option<String>,
+    pub subject: Option<String>,
+    pub disable_keygen: bool,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub async fn handle_par_create(
+    ParCreateArgs {
+        // claims related options
+        capid,
+        vendor,
+        revision,
+        version,
+        schema,
+        issuer,
+        subject,
+        name,
+        // par related options
+        directory,
+        arch,
+        binary,
+        destination,
+        compress,
+        disable_keygen,
+    }: ParCreateArgs,
+    output_kind: OutputKind,
+) -> Result<String> {
+    let mut par = ProviderArchive::new(&capid, &name, &vendor, revision, version);
+
+    let mut f = File::open(binary.clone())?;
+    let mut lib = Vec::new();
+    f.read_to_end(&mut lib)?;
+
+    let issuer = extract_keypair(
+        issuer,
+        Some(binary.clone()),
+        directory.clone(),
+        KeyPairType::Account,
+        disable_keygen,
+        output_kind,
+    )?;
+    let subject = extract_keypair(
+        subject,
+        Some(binary.clone()),
+        directory,
+        KeyPairType::Service,
+        disable_keygen,
+        output_kind,
+    )?;
+
+    par.add_library(&arch, &lib).map_err(convert_error)?;
+
+    let extension = if compress { ".par.gz" } else { ".par" };
+    let outfile = match destination {
+        Some(path) => path,
+        None => format!(
+            "{}{}",
+            PathBuf::from(binary.clone())
+                .file_stem()
+                .unwrap()
+                .to_str()
+                .unwrap(),
+            extension
+        ),
+    };
+    if let Some(ref schema) = schema {
+        let bytes = std::fs::read(schema)?;
+        par.set_schema(
+            serde_json::from_slice::<serde_json::Value>(&bytes)
+                .with_context(|| "Unable to parse JSON from file contents".to_string())?,
+        )
+        .map_err(convert_error)
+        .with_context(|| format!("Error parsing JSON schema from file '{:?}'", schema))?;
+    }
+
+    par.write(&outfile, &issuer, &subject, compress)
+        .await
+        .map_err(|e| anyhow!("{}", e))
+        .with_context(|| {
+            format!(
+                "Error writing PAR. Please ensure directory {:?} exists",
+                PathBuf::from(outfile.clone()).parent().unwrap(),
+            )
+        })?;
+    Ok(outfile)
+}
+
+pub async fn handle_par_insert(
```

Comment:
With the refactors we can rename this too
```suggestion
pub async fn insert_provider_binary(
```